File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.py
class Solution:
    def distinctAverages(self, nums: List[int]) -> int:
        nums.sort()
        i,j =0,len(nums)-1
        output = set()
        while(i<j):
            output.add((nums[i]+nums[j])/2)
            i+=1
            j-=1
        return len(output)

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
       int l = 0, r = nums.length-1;
	   HashSet<Double> res = new HashSet<>();
       Arrays.sort(nums);
       while(l<r){
           res.add((nums[l]+nums[r])/2.0);
           l++;
           r--;
       }
       return res.size();
    }
}